src/components/layout/index.js

```diff
@@ -0,0 +1,64 @@
+import React, {createContext} from 'react';
+import cx from 'classnames';
+import PropTypes from 'prop-types';
+
+const LayoutContext = createContext({
+  isSidebarVisible: true
+});
+
+export class Layout extends React.Component {
+  static displayName = 'Layout';
+
+  static propTypes = {
+    children: PropTypes.node,
+  }
+
+  state = {
+    isSidebarVisible: false
+  };
+
+  render() {
+    return (
+      <LayoutContext.Provider
+        value={ {
+          showSidebar: () => this.setState({ isSidebarVisible: true }),
+          hideSidebar: () => this.setState({ isSidebarVisible: false }),
+          isSidebarVisible: this.state.isSidebarVisible,
+        } }
+      >
+        <LayoutContext.Consumer>
+          {(context) => (
+            <div
+              className={ cx('layout-root', {
+                'sidebar-visible': context.isSidebarVisible,
```

Comment:
@gazdagergo In this component you don't have to use the consumer element because you can reach the state so it doesn't make any sense. The main goal of the context API that you can reach the grandparents' stuff from grandchildren and you don't have to pass your props through a complex component tree.

In our case with the layout component we want to give the chance to other components to control the visibility of the sidebar (which is the part of the layout) "remotely".

You should rather export the `LayoutContext` to let others to use its `Consumer` component to change the sidebar visibility. 

For example:

```js
// in the article page
import { LayoutContext } from '@pi/intellyo-components';

class ArticleEditor extends React.Component {
  componentDidMount() {
    this.props.layoutContext.hideSidebar();
  }
  componentWillUnMount() {
    this.props.layoutContext.showSidebar();
  }
}

class ArticleEditorWithLayoutContext extends React.Component {
  render() {
    <LayoutContext.Consumer>
      {
        (context) => (
          <ArticleEditor
            layoutContext={ context }
          />
        )
      }
    </LayoutContext.Consumer>
  }
}
```

Reply:
OK, I will check.